types/akamai-edgeworkers/README.md

```diff
@@ -0,0 +1,62 @@
+TypeScript types for the Akamai [EdgeWorker API]. This allows you to write 
+your EdgeWorkers in TypeScript.  
+
+Types are available for the `Request` and `Response` objects, as well as the 
+built-in modules.   
+
+# User Guide
+
+EdgeWorkers are written in ECMAScript6, so you need to set your 
+`tsconfig.json` to use `es6` as the compilation target and module 
+code generator:
+
+```json5
+{
+    "compilerOptions": {
+        "module": "es6",
+        "target": "es6",
+ //...
+    }
+}
+```    
+
+## Using the `Request` and `Response` Objects
+
+The predefined EdgeWorker callbacks take Request and Response objects as 
+arguments. After you have installed this package, you can create a `main.ts`
+with the following stubs:
+
+```typescript
+/// <reference types="akamai-edgeworkers-tsc"/>
+
+export function onClientRequest(request : EW.MutableRequest & EW.HasRespondWith){}
+export function onOriginRequest(request : EW.MutableRequest) {}
+export function onOriginResponse(request : EW.ImmutableRequest & EW.HasRespondWith, response : EW.Response) {}
+export function onClientResponse(request : EW.ImmutableRequest, response : EW.Response) {} 
+```
+
+The triple-slashed first line references this package and pulls `EW` into your 
+namespace. 
+
+## Using Built-In Modules
```

Comment:
Instead of using a `"paths"` mapping, you could move the contents of `cookies.d.ts` and `url-search-params.d.ts` into `index.d.ts` as ambient modules, similar to how NodeJS handles its module definitions. I've added more specifics about this in the relevant files.

Reply:
That's much better than messing with paths. Thanks for the tip!